### PR TITLE
NETOBSERV-1087: Added fields for ca certificate configuration

### PR DESCRIPTION
**Pull request by OlivierCazade**

### CODE REVIEW

File: controllers/flowlogspipeline/flp_common_objects.go [hunk 1]
```diff
@@ -742,9 +742,30 @@ func (b *builder) serviceMonitor() *monitoringv1.ServiceMonitor {
...
+				flpServiceMonitorObject.Spec.Endpoints[0].TLSConfig.SafeTLSConfig.CA = monitoringv1.SecretOrConfigMap{
+					Secret: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: b.desired.Processor.Metrics.Server.TLS.Provided.Name,
+						},
+						Key: b.desired.Processor.Metrics.Server.TLS.Provided.CaFile,
+					},
+				}
+			}
+		}
```
Comment (jpinsonneau) [inline controllers/flowlogspipeline/flp_common_objects.go:768]: Do we need to provide ConfigMap or Secret if `InsecureSkipVerify` is true ?

I would suggest to create [an helper](https://github.com/netobserv/network-observability-operator/blob/main/pkg/helper/flowcollector.go) to check both `b.desired.Processor.Metrics.Server.TLS.Provided.Type` and `b.desired.Processor.Metrics.Server.TLS.Provided.InsecureSkipVerify`

**Review by jotak — CHANGES_REQUESTED** [review]

File: controllers/flowlogspipeline/flp_common_objects.go (hunk 1 above)
Comment (OlivierCazade) [reply controllers/flowlogspipeline/flp_common_objects.go]: Done, thanks!

File: api/v1beta1/flowcollector_types.go [hunk 2]
```diff
@@ -678,6 +678,16 @@ type CertificateReference struct {
 	// certKey defines the path to the certificate private key file name within the config map or secret. Omit when the key is not necessary.
 	// +optional
 	CertKey string `json:"certKey,omitempty"`
+
```
[the remark this answers is not in the document]
Comment (OlivierCazade) [reply api/v1beta1/flowcollector_types.go]: I moved the field directly to the ServerTLS.

File: api/v1alpha1/flowcollector_webhook.go [hunk 3]
```diff
@@ -162,3 +165,11 @@ func Convert_v1beta1_FlowCollectorExporter_To_v1alpha1_FlowCollectorExporter(in
 func Convert_v1beta1_FlowCollectorEBPF_To_v1alpha1_FlowCollectorEBPF(in *v1beta1.FlowCollectorEBPF, out *FlowCollectorEBPF, s apiconversion.Scope) error {
 	return autoConvert_v1beta1_FlowCollectorEBPF_To_v1alpha1_FlowCollectorEBPF(in, out, s)
 }
+
+// // This function need to be manually created because conversion-gen not able to create it intentionally because
+// // we have new defined fields in v1beta1 not in v1alpha1
+// // nolint:golint,stylecheck,revive
+// func Convert_v1beta1_CertificateReference_To_v1alpha1_CertificateReference(in *v1beta1.CertificateReference, out *CertificateReference, s apiconversion.Scope) error {
```
Comment (jotak) [inline api/v1alpha1/flowcollector_webhook.go:172]: nit: looks like some garbage commented out?

Comment (OlivierCazade) [reply api/v1alpha1/flowcollector_webhook.go]: Yes, removed, thank you!

**Review by jotak — APPROVED** [review]

**Comment by jotak** [comment]
I'm merging for the branch cut, so this must be verified post-merge

**Pull request merged**